psique/src/App/test/wais/subtest/puzlesVisuales.js

```diff
@@ -1,8 +1,312 @@
+    var total = 0;
+    for(var i=2;i<results.length;i++){
+      total = total + results[i];
+    }
+    return total;
+  }
+
+  function nextDemoExample(key){
+    var punt = parseInt(key);
+    setResults(update(results,{
+      [numberItem]: {
+        $set: punt
+    }}))
+
+    if(!example){
+      example = true;
+      setNumberItem(numberItem+1);      
+
+    }else{
+      next();
+    }
+  }
+
+  function content(){
+    switch (state) {
+      case 'instruccion':
+        return (
+          <div id= "inicio" >
+            <h1>Pluzles Visuales</h1>
+            <b>Intrucciones generales:</b>
+            <p>Se presentará una imagen principal en la parte superior y seis imágenes más debajo de esta.</p>
+            <p>La tarea es escoger, entre las seis opciones, las tres figuras que combinadas forman la figura principal </p>
+            <br/>
+            <b>Intrucciones de calificacion:</b>
+            <p>Para calificar se debe presionar '1' para indicar una respuesta correcta por el paciente o '0' en caso de que sea incorrecta  </p>
+            <br/>
+            <CustomButton
+              msj="Iniciar subprueba"
+              callback={next}
+            ></CustomButton>  
+          </div>
+        )
+      
+      case "seleccion":
+        return(
+         <div>
+            <h1>Pluzles Visuales</h1>
+            <p>¿En que estimulo desea iniciar la prueba? </p>
+            <p>Pacientes con sospechas de discapacidad intelectual:</p>
+            <CustomButton
+              msj="Estimulo 1"
+              callback={()=>imagenInit(2)}
+            ></CustomButton>
+
+            <p>Pacientes de edad 16-89:</p>
+            <CustomButton
+              msj="Estimulo 5"
+              callback={()=>imagenInit(6)}
+            ></CustomButton> 
+          </div>
+        )
+
+      case "ejemplo-demostracion":
+        return(
+          <div >
+            <h1> Estimulo {example ? "Ejemplo": "Demostración"}</h1>
+            <img 
+              className={classes.img} 
+              alt={"Estimulo "+state+numberItem} 
+              src={require("../../../assets/estimulos/PuzlesVisuales/"+stimuliSrc[numberItem]+".jpg")}
+            />
+            
+            <KeyboardEventHandler 
+              handleKeys={['1','0']}
+              onKeyEvent={(key, e) => {
+                nextDemoExample(parseInt(key))
+                }}
+            />
+          </div>
+        )
+
+       case "test":
+         return(
+          <div>
+            <h1> Estimulo {numberItem-1}</h1>
+            <img 
+              className={classes.img}
+              alt={"Estimulo "+(numberItem-1)}
+              src={require("../../../assets/estimulos/PuzlesVisuales/"+stimuliSrc[numberItem]+".jpg")}
+            />
+
+            <KeyboardEventHandler 
+              handleKeys={['1','0']}
+              onKeyEvent={(key, e) => {
+                score(parseInt(key))
+              }}
+            />
+          </div>
+         )
```

Comment:
Here you can add the option to enter the patient answer and score the answer autimatically.